x/dymns/keeper/hooks.go

```diff
@@ -0,0 +1,370 @@
+package keeper
+
+import (
+	"errors"
+
+	"github.com/cometbft/cometbft/libs/log"
+
+	errorsmod "cosmossdk.io/errors"
+
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+
+	"github.com/osmosis-labs/osmosis/v15/osmoutils"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	dymnstypes "github.com/dymensionxyz/dymension/v3/x/dymns/types"
+	rollapptypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+	epochstypes "github.com/osmosis-labs/osmosis/v15/x/epochs/types"
+)
+
+/* -------------------------------------------------------------------------- */
+/*                              x/epochs hooks                                */
+/* -------------------------------------------------------------------------- */
+
+var _ epochstypes.EpochHooks = epochHooks{}
+
+type epochHooks struct {
+	Keeper
+}
+
+// GetEpochHooks returns the epoch hooks for the module.
+func (k Keeper) GetEpochHooks() epochstypes.EpochHooks {
+	return epochHooks{
+		Keeper: k,
+	}
+}
+
+// BeforeEpochStart is the epoch start hook.
+func (e epochHooks) BeforeEpochStart(_ sdk.Context, _ string, _ int64) error {
+	return nil
+}
+
+// AfterEpochEnd is the epoch end hook.
+func (e epochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	miscParams := e.MiscParams(ctx)
+
+	if epochIdentifier != miscParams.EndEpochHookIdentifier {
+		return nil
+	}
+
+	logger := e.Logger(ctx).With("hook", "After-Epoch-End", "epoch-number", epochNumber, "epoch-identifier", epochIdentifier)
+
+	if miscParams.EnableTradingName {
+		if err := e.processActiveDymNameSellOrders(ctx, logger); err != nil {
+			return err
+		}
+	}
+
+	if miscParams.EnableTradingAlias {
+		if err := e.processActiveAliasSellOrders(ctx, logger); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// processActiveDymNameSellOrders process the finished Dym-Name Sell-Orders.
+// Sell-Order will be deleted. If the Sell-Order has a winner, the Dym-Name ownership will be transferred.
+func (e epochHooks) processActiveDymNameSellOrders(ctx sdk.Context, logger log.Logger) error {
+	finishedSOs, aSoe := e.getFinishedSellOrders(ctx, dymnstypes.TypeName, logger)
+
+	if len(finishedSOs) < 1 || len(aSoe.Records) < 1 {
+		return nil
+	}
```

Comment:
is this necessary? or just micro optimization?

Reply:
This is hook, why not optimization?